econplayground/main/models.py

```diff
@@ -474,3 +474,87 @@ class Meta:
 
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class Question(models.Model):
+    adaptive = models.BooleanField(default=False)
+    ap_correct = models.ForeignKey(
+        'self',
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name='ap_continue'
+    )
+    ap_incorrect = models.ForeignKey(
+        'self',
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        related_name='ap_intervene'
+    )
+    title = models.CharField(max_length=256, default='Untitled', unique=True)
+
+    embedded_media = models.TextField(blank=True, default='')
+    graph = models.ForeignKey(
+        Graph, on_delete=models.CASCADE, blank=True)
+
+    is_assessment = models.BooleanField(default=True)
+    is_key = models.BooleanField(default=True)
+    prompt = models.TextField(blank=True, default='')
+    instructors = models.ManyToManyField(User)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def get_url_embed(self):
+        return self.embedded_media
+
+    def get_graph_name(self):
+        return self.graph.title
+
+    def __str__(self):
+        if len(self.prompt) < 33:
+            return self.prompt
+        else:
+            return self.prompt[0, 32] + '...'
+
+    def clone(self):
+        c = copy.copy(self)
+        c.pk = None
+        c.is_sample = None
+        c.title = self.title + '_copy'
+        c.save()
+
+        return c
+
+
+class Assignment(models.Model):
+    questions = models.ManyToManyField(Question, blank=True)
+    title = models.CharField(max_length=256, default='Untitled', unique=True)
+    cohorts = models.ManyToManyField(Cohort, blank=True)
+    instructors = models.ManyToManyField(User)
```

Comment:
ManyToMany fields can be difficult to work with, in practice. To keep it simple, let's just tie the Assignment to a single Instructor.

Reply:
So, this was something that came up at the Friday meeting with Tom (for questions at least, he might have also meant assignments, but I didn't write that down). He had 3 use cases that he wanted to cover:
1) Share - Where two instructors would be able to modify the same instance of a Question.
2) Copy - A clone feature to create an independent copy of a Question-- I just took the cohort clone and modified it.
3) View - A read-only share.
I can change the model type, but either way I think we will need some way to model permissions. I guess have the original owner as the instructor and some list of permitted instructors to share with?

Reply:
I think just tying the Assignment to a single Instructor is best for now - allowing all instructors to view all assignments in the bank. As I mentioned, Tom is really the only instructor using this, and even if there are more in the future, there are other ways for handling this permission-type functionality. In any case, that stuff is lower priority than just getting the basics working for Tom.

Reply:
Understood. Priorities on the MVP we can refactor afterwards.